2020-06-01-DANCE-1-eMolecules-t142-selected/02_create_torsiondrive_dataset.py

```diff
@@ -0,0 +1,149 @@
+
+
+def read_optimization_inputs(input_json_gz):
+    """ Read data generated by 01_generate.py
+    Returns
+    -------
+    optimization_inputs: dict
+        Dictionary for selected torsions, has this structure:
+        [
+            {
+                'cmiles_identifiers': {
+                    "canonical_smiles": ...,
+                    ..
+                },
+                'atom_indices': [
+                    [
+                        1,
+                        2,
+                        3,
+                        4
+                    ]
+                ],
+                'initial_molecules': [
+                    Molecule1a,
+                    Molecule1b,
+                    ..
+                ]
+            },
+            ..
+        ]
+    """
+    with gzip.open(input_json_gz, 'r') as f:
+        optimization_inputs = json.load(f)
+    return optimization_inputs
+
+
+def main():
+    """Loads the input and runs all the calculations."""
+    print("Reading optimization_inputs...")
+    optimization_inputs = read_optimization_inputs(INPUT_FILE)
+    print(f"Found {len(optimization_inputs)} torsions")
+
+    print("Initializing dataset...")
+    #  client = ptl.FractalClient("localhost:7777", verify=False)
+    client = ptl.FractalClient.from_file()
+
+    if UPDATE:
+        dataset = client.get_collection("TorsionDriveDataset", COLLECTION_NAME)
+    else:
+        # create a new dataset with specified name
+        dataset = ptl.collections.TorsionDriveDataset(COLLECTION_NAME,
+                                                      client=client)
+
+        # create specification for this dataset
+        opt_spec = {
+            "program": "geometric",
+            "keywords": {
+                "coordsys": "tric",
+                "enforce": 0.1,
+                "reset": True,
+                "qccnv": True,
+                "epsilon": 0.0,
+            },
+        }
+        keyword = ptl.models.KeywordSet(
+            values={
+                "maxiter":
+                    200,
+                "scf_properties": [
+                    "dipole",
+                    "quadrupole",
+                    "wiberg_lowdin_indices",
+                    "mayer_indices",
+                ],
+            })
+        keyword_id = client.add_keywords([keyword])[0]
+
+        qc_spec = {
+            "driver": "gradient",
+            "method": "B3LYP-d3bj",
+            "basis": "dzvp",
+            "program": "psi4",
+            "keywords": keyword_id,
+        }
+        dataset.add_specification(
+            "default",
+            opt_spec,
+            qc_spec,
+            description="Standard OpenFF torsiondrive specification.",
+        )
+
+    # Add molecules.
+    print(f"Adding {len(optimization_inputs)} torsions")
+    i = 0
+
+    for item in tqdm.tqdm(optimization_inputs):
+        attributes = item["cmiles_identifiers"]
+        canonical_torsion_index = attributes["canonical_isomeric_smiles"]
```

Comment:
I put this as a temporary value; if I have the canonical SMILES and a list of torsion indices, what is the correct way to do this?